TrigScint/include/TrigScint/Firmware/hitproducer.h

```diff
@@ -0,0 +1,13 @@
+#ifndef HITPRODUCER_H
+#define HITPRODUCER_H
+
+#include "objdef.h"
+
+void copyHit1(Hit One, Hit Two);
+void copyHit2(Hit One, Hit Two);
+void hitproducer_ref(ap_uint<14> FIFO[NHITS][5], Hit outHit[NHITS],
+                     ap_uint<8> Peds[NHITS]);
```

Comment:
Similar comment to the `pragma` comment. These are not defined in the source file and so they will lead to an error. Should we wrap them in some compiler definition `#ifdef` to avoid them when we are emulating (here) and keep them for when this file is not being used for emulation (somewhere else?).

Reply:
These are old pieces of code that used to mean something. The hitproducer_ref might be something HLS requires defined however, so while the copyHit1 may not need to be there in some capacity the _ref will be. So here I think the ifdef solution is reasonable.  I will investigate whether in vitis HLS there is a special bool somewhere that they use to enable the reading of pragmas; doubt it but I will check. Ruckus uses a cmake file unlike vitis HLS's standalone distribution, so if nothing else I can put a copy of whatever convention we adopt here in there.

Reply:
Done!